client/src/features/action/needs/NeedFormGoods.tsx

```diff
@@ -55,8 +57,13 @@ const initialFormData: ShareANeedData = {
 
 function NeedForm() {
   const [formData, setFormData] = React.useState<ShareANeedData>(initialFormData);
+  const [showPrompt, setShowPrompt] = React.useState(false);
```

Comment:
let's type the state `React.useState<boolean>(false);`

Reply:
this state doesn't define whether we're showing the prompt so i'd say naming should be something more like `isFormStarted` or `isFormInProgress`

Reply:
updated to isFormInProgress